daemon/daemon_unix_test.go

```diff
@@ -376,3 +379,75 @@ func sysInfo(t *testing.T, opts ...func(*sysinfo.SysInfo)) sysinfo.SysInfo {
 	}
 	return si
 }
+
+func TestGetBlkioWeightDevices(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "tempDevDirForBlkioThrottleDevices")
+	assert.NilError(t, err, "create temp file")
+
+	tempFile := filepath.Join(tempDir, "dev")
+
+	defer os.RemoveAll(tempDir)
+
+	// mock a file with major 0x1FD(509 in decimal) and minor 0x130(304)
+	if err = unix.Mknod(tempFile, unix.S_IFCHR, 0x11FD30); err != nil {
+		t.Fatalf("mknode error %s(%x): %v", tempFile, 0x11FD30, err)
+	}
+
+	mockResource := containertypes.Resources{
+		BlkioWeightDevice: []*blkiodev.WeightDevice{{Path: tempFile, Weight: 1024}},
+	}
+
+	weightDevs, err := getBlkioWeightDevices(mockResource)
+	assert.NilError(t, err, "getBlkioWeightDevices")
+
+	if len(weightDevs) != 1 {
```

Comment:
```go
assert.Check(t, is.Len(weightDevs, 1), "getBlkioWeightDevices")
```

(you can use `assert` in other places in your code as well, it makes the code more compact and perhaps more readable)